### ISSUE TITLE
CMakePresets.json no longer supports comments

### ISSUE THREAD

**Issue by alex1234567890123456789** [microsoft/vscode-cmake-tools#2169]
### Brief Issue Summary

In 1.7.3, you could use comments ( // ) in CMakePresets.json and everything worked fine.  Starting with 1.8.0, cmake tools fails to parse CMakePresets.json if it has comments in it.  While comments aren't officially allowed in json, many tools allow it, and it's not hard to support.  It is also REALLY helpful to be able to document your presets file!

### Debug Log

```
[presetController] Failed to parse CMakePresets.json: 
	SyntaxError: Unexpected token / in JSON at position 552
	at JSON.parse (<anonymous>)
	at PresetsController.parsePresetsFile (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:57684:32)
	at PresetsController.resetPresetsFile (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:57070:38)
	at async PresetsController.reapplyPresets (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:57085:9)
	at async watchPresetsChange (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:57015:13)
	at async Function.init (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:57043:9)
	at async Function.init (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:53142:35)
	at async CMakeToolsFolderController._addFolder (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:53333:22)
	at async CMakeToolsFolderController.loadAllCurrent (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:53285:17)
	at async ExtensionManager._init (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:51781:13)
	at async Function.create (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:51832:9)
	at async setup (/home/nvidia/.vscode-server/extensions/ms-vscode.cmake-tools-1.8.1/dist/main.js:52860:32)
	at async Promise.all (index 0)
	at async Promise.all (index 26)
	at async Promise.all (index 1)
```

**Comment by bobbrow** [comment]
We actually supported it before, but according to CMake release notes, it is not supposed to be supported: https://cmake.org/cmake/help/latest/release/3.19.html#id6

A bug was opened against us to remove the support and we did it because it matched the documented behavior: #2035

It's a [bug in CMake](https://gitlab.kitware.com/cmake/cmake/-/issues/22602) that comments are currently allowed.  We were trying to be helpful, but it turns out CMake accepts comments that are not at the top of the JSON file because there's a bug in the library they use. 🤦‍♂️

**Comment by alex1234567890123456789** [comment]
Thanks @bobbrow - can it be added back with a setting to enable/disable it?  It's REALLY really helpful...

**Comment by bobbrow** [comment]
I'm actually happy to go either way with this.  If folks want to advocate to Kitware that JSONC should be accepted, it would probably help if folks could comment on the issue I linked above.

We can also consider a setting to allow the comments in the meantime.

**Comment by bobbrow** [comment]
I opened a PR to add a setting for this.  We can try to push it into 1.9 (which we intended to release today) if there's time or I can build a separate VSIX shortly after.

**Comment by bobbrow** [comment]
The setting will be released with 1.9 hopefully later today.
